analyzers/src/SonarAnalyzer.Common/Rules/ParameterNameMatchesOriginalBase.cs

```diff
@@ -60,11 +60,11 @@ public abstract class ParameterNameMatchesOriginalBase<TSyntaxKind, TMethodDecla
                         }
                         else if (methodSymbol.OverriddenMethod != null)
                         {
-                            VerifyParameters(c, methodSyntax, methodSymbol.OverriddenMethod.Parameters, "base class");
+                            VerifyGenericParameters(c, methodSyntax, methodSymbol.OverriddenMethod.OriginalDefinition.Parameters, "base class");
```

Comment:
Please add a few test cases with deeper inheritance, just to be sure that the originalDefinition does the work in more "complex" cases.